### PR TITLE
SUP-19612: change Zoom video upload to add categoryEntry instead of entry->cateogy which is deprecated

### PR DESCRIPTION
**Pull request by amiras89**
This change is [Reviewable](https://reviewable.io/reviews/kaltura/server/9167)

### CODE REVIEW

File: plugins/vendor/zoom/lib/model/kZoomEngine.php [hunk 1]
```diff
@@ -353,11 +353,18 @@ protected function createEntryDescriptionFromMeeting($meeting)
...
 		{
-			$entry->setCategories($zoomIntegration->getZoomCategory());
+			$categoryEntry = new categoryEntry();
+			$categoryEntry->setEntryId($entry);
+			$categoryEntry->setCategoryId($zoomIntegration->getZoomCategoryId());
+			$categoryEntry->setPartnerId($entry->getPartnerId());
+			$categoryEntry->setCreatorKuserId($entry->getKuserId());
+			$categoryEntry->setStatus(CategoryEntryStatus::ACTIVE);
+			$categoryEntry->save();
+			KalturaLog::info("Entry Id '" . $entry->getId() ."' added to Zoom category Id " . $zoomIntegration->getZoomCategoryId());
```
Comment (MosheMaorKaltura) [inline plugins/vendor/zoom/lib/model/kZoomEngine.php]: If a string does not process values , don't use " prefer single quote '

Comment (amiras89) [reply plugins/vendor/zoom/lib/model/kZoomEngine.php]: Changed to single quotes

File: plugins/vendor/zoom/lib/model/kZoomEngine.php [hunk 2]
```diff
@@ -353,11 +353,18 @@ protected function createEntryDescriptionFromMeeting($meeting)
 	 * @param entry $entry
 	 * @throws kCoreException
 	 */
-	protected function setEntryCategory($zoomIntegration, $entry)
+	protected function createCategoryEntry($zoomIntegration, $entry)
```
Comment (MosheMaorKaltura) [inline plugins/vendor/zoom/lib/model/kZoomEngine.php]: This code should move to categoryentry core object
Adding new funino categoryEntry::add(entry, category)

When it is added , you must behave the same way cateogryEntryService::add handles the moderation.

File: alpha/lib/model/categoryEntry.php [hunk 3]
```diff
@@ -219,5 +219,120 @@ public function setCreatorPuserId ($v)
...
+	public function add($entryId, $categoryId)
+	{
+		/*=========================================================*/
+		$entry = entryPeer::retrieveByPK($entryId);
+		if(!$entry)
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_ENTRY_ID, $entryId);
+		}
+
+		$category = categoryPeer::retrieveByPK($categoryId);
```
Comment (MosheMaorKaltura) [inline alpha/lib/model/categoryEntry.php]: The entire function must be broken to smaller parts - 
function validateMaxCategoryEntries
function validateAccess 
function validateNotExist
function handleModeration
function assignUsers

**Pull request closed without merging**